app/workers/builds.php

```diff
@@ -0,0 +1,169 @@
+    {
+        $dbForProject = $this->getProjectDB($projectId);
+        
+        $function = $dbForProject->getDocument('functions', $functionId);
+        if ($function->isEmpty()) {
+            throw new Exception('Function not found', 404);
+        }
+
+        $deployment = $dbForProject->getDocument('deployments', $deploymentId);
+        if ($deployment->isEmpty()) {
+            throw new Exception('Deployment not found', 404);
+        }
+
+        $runtimes = Config::getParam('runtimes', []);
+        $key = $function->getAttribute('runtime');
+        $runtime = isset($runtimes[$key]) ? $runtimes[$key] : null;
+        if (\is_null($runtime)) {
+            throw new Exception('Runtime "' . $function->getAttribute('runtime', '') . '" is not supported');
+        }
+
+        $buildId = $deployment->getAttribute('buildId', '');
+        $build = null;
+        $startTime = \time();
+        if (empty($buildId)) {
+            $buildId = $dbForProject->getId();
+            $build = $dbForProject->createDocument('builds', new Document([
+                '$id' => $buildId,
+                '$read' => [],
+                '$write' => [],
+                'startTime' => $startTime,
+                'deploymentId' => $deploymentId,
+                'status' => 'processing',
+                'outputPath' => '',
+                'runtime' => $function->getAttribute('runtime'),
+                'source' => $deployment->getAttribute('path'),
+                'sourceType' => App::getEnv('_APP_STORAGE_DEVICE', Storage::DEVICE_LOCAL),
+                'stdout' => '',
+                'stderr' => '',
+                'endTime' => 0,
+                'duration' => 0
+            ]));
+            $deployment->setAttribute('buildId', $buildId);
+            $deployment = $dbForProject->updateDocument('deployments', $deploymentId, $deployment);
+        } else {
+            $build = $dbForProject->getDocument('builds', $buildId);
+        }
+
+        /** Request the executor to build the code... */
+        $build->setAttribute('status', 'building');
+        $build = $dbForProject->updateDocument('builds', $buildId, $build);
+
+        $source = $deployment->getAttribute('path');
+        $vars = $function->getAttribute('vars', []);
+        $baseImage = $runtime['image'];
+
+        try {
+            $response = $this->executor->createRuntime(
+                projectId: $projectId, 
+                functionId: $functionId, 
+                deploymentId: $deploymentId, 
+                source: $source,
+                destination: APP_STORAGE_BUILDS . "/app-$projectId",
+                vars: $vars, 
+                runtime: $key, 
+                baseImage: $baseImage,
+                workdir: '/usr/code',
+                remove: true,
+                commands: [
+                    'sh', '-c',
+                    'tar -zxf /tmp/code.tar.gz -C /usr/code && \
+                    cd /usr/local/src/ && ./build.sh'
+                ]
```

Comment:
Why are we not sending entrypoint here? createRuntime requires entrypoint which is used by runtimes.